keepassxc-browser/content/autocomplete.js

```diff
@@ -131,45 +146,85 @@ class Autocomplete {
         for (const c of this.elements) {
-            const item = document.createElement('div');
-            item.textContent = c.label;
-            item.setAttribute('uuid', c.uuid);
+            const item = this.generateItem(c);
 
-            const itemInput = kpxcUI.createElement('input', '', { 'type': 'hidden', 'value': c.value });
+            const itemInput = kpxcUI.createElement('input', '', { type: 'hidden', value: c.value });
             item.append(itemInput);
             item.addEventListener('click', e => this.itemClick(e, this.input, c.uuid));
 
             // These events prevent the double hover effect if both keyboard and mouse are used
             item.addEventListener('mousemove', e => this.mouseMove(e));
-
             item.addEventListener('mousedown', e => e.stopPropagation());
             item.addEventListener('mouseup', e => e.stopPropagation());
 
             // If this page has an associated uuid and it matches this credential, then put it on top of the list
-            if (username === c.value
-                || (this.afterFillSort === SORT_BY_RELEVANT_ENTRY && c.uuid === pageUuid)) {
+            if (username === c.value || (this.afterFillSort === SORT_BY_RELEVANT_ENTRY && c.uuid === pageUuid)) {
                 this.list.prepend(item);
             } else {
                 this.list.appendChild(item);
             }
         }
 
+        // Use only two columns if groups are not shown
+        if (this.useCompactMode && !this.showGroup) {
+            this.list.style.gridTemplateColumns = 'repeat(2, auto)';
+        }
+
         this.selectItem();
     }
 
+    // Add parentElement parameter
+    generateItem(credential) {
+        // Create the DOM element for the list item wrapper.
+        const item = kpxcUI.createElement('div', 'kpxcAutocomplete-item', {
+            title: `${credential.title} | ${credential.value} | ${credential.group}`
+        });
+        item.setAttribute('uuid', credential.uuid);
+
+        // Create the DOM element for the list item header.
+        const itemHeader = document.createElement('div');
+        itemHeader.classList.add('kpxcAutocomplete-item__header');
+
+        // Create the DOM element for showing the group only when the group name is available.
+        if (!this.useCompactMode) {
+            const itemGroup = kpxcUI.createElement('div', 'kpxcAutocomplete-item__group', {}, credential.group);
+            itemHeader.append(itemGroup);
+        }
+
+        // Create the DOM element for showing the credentials name.
+        const itemLabel = kpxcUI.createElement('div', 'kpxcAutocomplete-item__label', {}, credential.title);
+        itemHeader.append(itemLabel);
+
+        // Append the header to the list item.
+        item.append(itemHeader);
+
+        // Create the DOM element for showing the credentials username field.
+        const itemClassName = this.showGroup ? 'kpxcAutocomplete-item__value' : 'kpxcAutocomplete-item__value_last';
```

Comment:
We are using `__` to separate DOM parts, but there is a single `_` for last, is that intentional? We also use `--` a lot, this is just style, but we should be consistent. I think this one should be `kpxcAutocomplete-item__value--last`

Reply:
I didn't touch these at all. Let's modify them a bit.